consoleme/lib/dynamo.py

```diff
@@ -466,12 +466,23 @@ async def write_policy_request_v2(self, extended_request: ExtendedRequestModel):
             "extended_request": json.loads(extended_request.json()),
             "username": extended_request.requester_email,
         }
+
+        if extended_request.principal.principal_type == "AwsResource":
```

Comment:
Structural pattern matching in py3.10 will be rad for a few of these patterns we have

Reply:
Yesss, I keep forgetting this is coming